Core/Core/Grades/Grades.swift

```diff
@@ -0,0 +1,293 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2018-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import CoreData
+
+private let UnknownGradingPeriodID = "unknown"
+
+public class Grades {
+    let courseID: String
+    let userID: String?
+    var env: AppEnvironment {
+        return .shared
+    }
+    var gradingPeriodID: String? = UnknownGradingPeriodID {
+        didSet {
+            reload()
+        }
+    }
+    var callback: (() -> Void)?
+    var error: Error?
+    private var pending = 0 {
+        didSet {
+            if oldValue <= 0 && pending > 0 {
+                notify()
+            } else if oldValue > 0 && pending <= 0 {
+                notify()
+            }
+        }
+    }
+    var isPending: Bool { pending > 0 }
+
+    var assignments: Store<LocalUseCase<Assignment>>!
+    var gradingPeriods: Store<LocalUseCase<GradingPeriod>>!
+    var enrollments: Store<LocalUseCase<Enrollment>>!
```

Comment:
do these need to be forceUnwrapped.  Looks like you set them in the init.

Reply:
Their initializers use `self` (`[weak self]` for the callback) so they have to either be optional or force unwrapped.